larky/src/main/resources/stdlib/xml/etree/ElementTree.star

```diff
@@ -941,72 +955,68 @@ def _namespaces(elem, default_namespace=None):
                 add_qname(tag)
         elif tag != None and tag != Comment and tag != PI:
             _raise_serialization_error(tag)
-        for key, value in elem.items():
+        for key, value in current.items():
             if types.is_instance(key, QName):
                 key = key.text
             if key not in qnames:
                 add_qname(key)
             if types.is_instance(value, QName) and value.text not in qnames:
                 add_qname(value.text)
-        text = elem.text
+        text = current.text
         if types.is_instance(text, QName) and text.text not in qnames:
             add_qname(text.text)
+        qu.extend(current._children)
     return qnames, namespaces
 
+def prepare_elems_for_serialize(tops_level_elems):
+    new_elems = []
+    for el in tops_level_elems:
+        new_elems.append(el)
+        qu = el._children[0:]
```

Comment:
This is a good catch, so we have to do `list[0:]` to copy the list instead of python `list[:]`, so it is *intentional* to put the 0 before the `:` (colon).